lib/runtime/wasmtime/legacy_imports.go

```diff
@@ -541,200 +541,128 @@ func ext_ed25519_sign(c *wasmtime.Caller, idData, pubkeyData, msgData, msgLen, o
-		ext_ed25519_verify.AsExtern(),
-		ext_sr25519_verify.AsExtern(),
-		ext_sandbox_memory_get.AsExtern(),
-		ext_sandbox_memory_set.AsExtern(),
-		ext_blake2_256_enumerated_trie_root.AsExtern(),
-		ext_malloc.AsExtern(),
-		ext_free.AsExtern(),
-		ext_twox_64.AsExtern(),
-		ext_log.AsExtern(),
+// ImportLegacyNodeRuntime add wasmtime imports to linker
+func ImportLegacyNodeRuntime(store *wasmtime.Store) (*wasmtime.Linker, error) {
+	linker := wasmtime.NewLinker(store)
+
+	fns := []struct {
+		name string
+		fn   interface{}
+	}{
+		{"ext_blake2_256", ext_blake2_256},
+		{"ext_malloc", ext_malloc},
+		{"ext_print_num", ext_print_num},
+		{"ext_print_utf8", ext_print_utf8},
+		{"ext_print_hex", ext_print_hex},
+		{"ext_get_storage_into", ext_get_storage_into},
+		{"ext_set_storage", ext_set_storage},
+		{"ext_set_child_storage", ext_set_child_storage},
+		{"ext_storage_root", ext_storage_root},
+		{"ext_storage_changes_root", ext_storage_changes_root},
+		{"ext_get_allocated_storage", ext_get_allocated_storage},
+		{"ext_clear_storage", ext_clear_storage},
+		{"ext_clear_prefix", ext_clear_prefix},
+		{"ext_blake2_256_enumerated_trie_root", ext_blake2_256_enumerated_trie_root},
+		{"ext_twox_64", ext_twox_64},
+		{"ext_twox_128", ext_twox_128},
+		{"ext_sr25519_generate", ext_sr25519_generate},
+		{"ext_sr25519_public_keys", ext_sr25519_public_keys},
+		{"ext_sr25519_sign", ext_sr25519_sign},
+		{"ext_sr25519_verify", ext_sr25519_verify},
+		{"ext_ed25519_generate", ext_ed25519_generate},
+		{"ext_ed25519_verify", ext_ed25519_verify},
+		{"ext_is_validator", ext_is_validator},
+		{"ext_local_storage_get", ext_local_storage_get},
+		{"ext_local_storage_compare_and_set", ext_local_storage_compare_and_set},
+		{"ext_network_state", ext_network_state},
+		{"ext_submit_transaction", ext_submit_transaction},
+		{"ext_local_storage_set", ext_local_storage_set},
+		{"ext_kill_child_storage", ext_kill_child_storage},
+		{"ext_sandbox_memory_new", ext_sandbox_memory_new},
+		{"ext_sandbox_memory_teardown", ext_sandbox_memory_teardown},
+		{"ext_sandbox_instantiate", ext_sandbox_instantiate},
+		{"ext_sandbox_invoke", ext_sandbox_invoke},
+		{"ext_sandbox_instance_teardown", ext_sandbox_instance_teardown},
+		{"ext_get_allocated_child_storage", ext_get_allocated_child_storage},
+		{"ext_child_storage_root", ext_child_storage_root},
+		{"ext_clear_child_storage", ext_clear_child_storage},
+		{"ext_secp256k1_ecdsa_recover_compressed", ext_secp256k1_ecdsa_recover_compressed},
+		{"ext_sandbox_memory_get", ext_sandbox_memory_get},
+		{"ext_sandbox_memory_set", ext_sandbox_memory_set},
+		{"ext_log", ext_log},
+		{"ext_free", ext_free},
+	}
+	for _, f := range fns {
+		if err := linker.DefineFunc("env", f.name, f.fn); err != nil {
+			return nil, err
+		}
+	}
+	return linker, nil
+}
+
+// ImportHostHandler adds the wasmtime imports for host API tester to linker
+func ImportHostHandler(store *wasmtime.Store, linker *wasmtime.Linker) error {
```

Comment:
could you update this to also match `func(store *wasmtime.Store) (*wasmtime.Linker, error)`?